alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/processor/service/ProcessorServiceImpl.java

```diff
@@ -12,23 +12,31 @@
 public class ProcessorServiceImpl implements ProcessorService {
     private static final Logger logger = LoggerFactory.getLogger(ProcessorServiceImpl.class);
 
-    @Getter
-    @Setter
+    @Getter @Setter
     protected RetryingTransactionHelper retryingTransactionHelper;
 
     @Getter @Setter
     protected ExecutorService executorService = Executors.newFixedThreadPool(5);
 
+    @Getter @Setter
+    protected HealthProcessorConfiguration configurationService;
+
     protected  ProcessorServiceImpl() {}
```

Comment:
Why is this constructor needed? Will it be used?
Can we just make this an `@Data` class with an `@AllArgsConstructor` and drop the separate `@Getter` and `@Setter` annotations and the explicit constructors?

Reply:
don't use `@Data` for services

Reply:
but :+1: for constructor injection and dropping unnecessary `@Getter` and `@Setter` annotations

Reply:
I'd still like to keep `@Setter(AccessLevel.PACKAGE)` to be able to only inject the mocked services?

Reply:
... and `@NoArgsConstructor(access = AccessLevel.PACKAGE)` to be able to use `@Spy` ?

Reply:
1. use constructor injection for the mocks
2. use `Mockito.spy(instance)` to wrap your instance with spy-instrumentation

Reply:
After discussion with Daan I already dropped these